backend/lib/edgehog/triggers/handler/manual_actions/handle_trigger.ex

```diff
@@ -349,16 +377,17 @@ defmodule Edgehog.Triggers.Handler.ManualActions.HandleTrigger do
 
         with {:ok, deployment} <- Containers.fetch_deployment(deployment_id, tenant: tenant) do
           case state do
-            nil ->
-              Containers.delete_deployment(deployment)
-
             "Started" ->
               Containers.mark_deployment_as_started(deployment, tenant: tenant)
               Containers.deployment_update_resources_state(deployment, tenant: tenant)
 
             "Stopped" ->
               Containers.mark_deployment_as_stopped(deployment, tenant: tenant)
               Containers.deployment_update_resources_state(deployment, tenant: tenant)
+
+            # The device is removing the deployment, remove it!
+            nil ->
+              Containers.destroy_deployment!(deployment, tenant: tenant)
```

Comment:
Contrary to the previous comment, here the the value of the function calls are directly used as the return value of the event handler.
Since the `handle_event` function should return an `{:ok, value}` tuple, we should respect that here and avoid using the `!` function variant as the last function call. Otherwise, we produce a match exception within the `run/3` function that will return a 500 error to the API client.

```suggestion
          case state do
            "Started" ->
              Containers.mark_deployment_as_started!(deployment, tenant: tenant)
              |> Containers.deployment_update_resources_state(tenant: tenant)

            "Stopped" ->
              Containers.mark_deployment_as_stopped!(deployment, tenant: tenant)
              |> Containers.deployment_update_resources_state(tenant: tenant)

            # The device is removing the deployment, remove it!
            nil ->
              Containers.destroy_deployment(deployment, tenant: tenant)
          end
```